rpc/types/events.go

```diff
@@ -97,40 +83,38 @@ func ParseTxResult(result *abci.ResponseDeliverTx) (*ParsedTxs, error) {
 		TxHashes: make(map[common.Hash]int),
 	}
 	for _, event := range result.Events {
```

Comment:
the nested if-elses in here get a bit complex, not sure if it can be improved to be more readable

Reply:
this part is indeed a bit tricky because it tries to be compatible with two versions of formats, I don't know how to simplify it further.